### ISSUE TITLE
Sidebar submenus do not collapse 

### ISSUE THREAD

**Issue by brumiser1550** [creativetimofficial/ct-material-dashboard-pro-angular#330]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I'm reporting the issue to the correct repository (for multi-repository projects)

# Expected Behavior

The sidebar submenu should collapse when clicked while open

# Current Behavior

The sidebar submenu does not collapse once open. It seems to get stuck in the "collapsing" state and never moves to the "collapse show" state. 

## Steps to Reproduce

1. Visit https://demos.creative-tim.com/material-dashboard-pro-angular2/#/dashboard
2. Open a submenu
3. Try and close a submenu

## Context

* Device: PC
* Operating System: Windows 10
* Browser and Version: FF and Chrome latest 

P.S. Sorry for sending an email, I didn't pay attention to the line that said "If this is a BUG report, please use the bellow OPEN GITHUB ISSUE button!"

**Comment by chelaruc** [comment]
Hi @brumiser1550. Thank you for using our product. I have tested on windows and it's working good on our end. Can you send me a video with the error?

**Comment by brumiser1550** [comment]
Same thing happens on FF and Chrome
https://drive.google.com/file/d/12UAO-yVM5xakTDKHV3pYNCkyVistP6_9/view

Google Chrome is up to date
Version 75.0.3770.100 (Official Build) (64-bit)

FireFox 68.0b11 (64-bit)

[image: image.png]

Thanks,
Brandon Rumiser
brumiser1550@gmail.com
770-616-1982

**Comment by chelaruc** [comment]
@brumiser1550 can you test the product on Chrome + incognito and tell me if it works.

**Comment by brumiser1550** [comment]
Same thing in Chrome and FF incognito mode. 

I have tested on my phone and another laptop so I think it may just be my laptop but not sure why.

**Comment by aarronte86** [comment]
Hi I'm having this same issue, and as @brumiser1550 it happens just in some clients but I couldn't find the pattern for the problem. Two clients with identical Chrome version and OS, in one the error happens but in the other one don't.

**Comment by chwernerstech** [comment]
Same Here, we have the issue in some PC with Chrome!

**Comment by chelaruc** [comment]
Do you get errors in the browser console?

**Comment by brumiser1550** [comment]
Nope, no console errors or anything running both in production and development mode.

**Comment by brumiser1550** [comment]
I was curious and opened some other demos and the first one with submenus works for me but the pro version still does not work. Working Demo -> https://demos.creative-tim.com/light-bootstrap-dashboard-pro-angular2/#/dashboard

**Comment by chwernerstech** [comment]
@chelaruc Please we need your help solving this, have you find something?

**Comment by chelaruc** [comment]
@chwernerstech Unfortunately, I can't reproduce the error.

**Comment by chwernerstech** [comment]
@chelaruc I can give you access to a pc that have the problem, let me know or write me to chwerner@gmail.com, many thanks!

**Comment by chelaruc** [comment]
@chwernerstech what version of Chrome do you have?

**Comment by chwernerstech** [comment]
Versión 75.0.3770.100 (Build oficial) (64 bits)
Could be the java or something?

**Comment by chwernerstech** [comment]
@chelaruc 
Versión 75.0.3770.100 (Build oficial) (64 bits)

**Comment by chelaruc** [comment]
Hi, @chwernerstech. Can you test if this happens on https://demos.creative-tim.com/material-dashboard-pro/examples/dashboard.html? Also, can you test here: https://getbootstrap.com/docs/4.3/components/collapse/?

**Comment by chwernerstech** [comment]
Hi @chelaruc , thanks for the reply.

The Link that you sent us is working, please could you tell us what to do?

Many Thanks!

WORKING
https://demos.creative-tim.com/material-dashboard-pro/examples/dashboard.html

NOT WORKING
https://demos.creative-tim.com/material-dashboard-pro-angular2/#/dashboard

**Comment by brumiser1550** [comment]
I tested https://demos.creative-tim.com/material-dashboard-pro/examples/dashboard.html and it works for me.

**Comment by chwernerstech** [comment]
@chelaruc How are you? Please could you tell us what to do ?

Please share with us the solution.

https://demos.creative-tim.com/material-dashboard-pro/examples/dashboard.html 

Many Thanks!!!

**Comment by michaelvivar** [comment]
@chelaruc I have the same problem using Edge or Internet Explorer. Could tell me how to fix it? Thanks!

**Comment by chelaruc** [comment]
@michaelvivar what errors do you get?

**Comment by michaelvivar** [comment]
No any errors in console, it just don't collapse the sidenav.
I've also tried to load the demo site (https://demos.creative-tim.com/material-dashboard-pro-angular2/#/dashboard) in Edge and IE, also not working.

**Comment by chwernerstech** [comment]
@chelaruc Please could you share the solution?

WORKING
https://demos.creative-tim.com/material-dashboard-pro/examples/dashboard.html

NOT WORKING
https://demos.creative-tim.com/material-dashboard-pro-angular2/#/dashboard

**Comment by aarronte86** [comment]
@chelaruc We are still having the same problem, we are looking forward to a solution.

**Comment by chelaruc** [comment]
@aarronte86 @chwernerstech @michaelvivar can you, guys, make a screen recording with the live demo and the browser console?

**Comment by chwernerstech** [comment]
@chelaruc Please look, we cant open the menu

http://www.izytech.com/video.mp4

**Comment by chwernerstech** [comment]
@chelaruc  anything that we can do? have a lot of troubles with the users!

**Comment by chelaruc** [comment]
@chwernerstech sorry for the late response. Can you change the bootstrap-material-design import like in this issue: https://github.com/creativetimofficial/ct-material-dashboard-pro-angular/issues/339?

**Comment by michaelvivar** [comment]
Finally. Thanks!

**Comment by brumiser1550** [comment]
That did not work for me, still cannot collapse sidebar menus. I have also had jQuery available in the console regardless. You got me thinking though so I started taking different scripts out of the angular.json that had to do with jquery and when I removed this line, with no other changes, everything worked. 

```
"node_modules/jqvmap/dist/maps/jquery.vmap.world.js",
```

**Comment by aarronte86** [comment]
@chelaruc your solution didn't work for me either, nor the @brumiser1550 change suggestion.  I had to make my own collapsible menu implementation using angular-material accordion component as an alternative while a solution was found.

**Comment by sebas0205** [comment]
Hi @chelaruc  , I´m still have this problem.  I did all the solutions here  but nothing .
I proved #339  and dont work for me
I proved too adn nothing : 

> That did not work for me, still cannot collapse sidebar menus. I have also had jQuery available in the console regardless. You got me thinking though so I started taking different scripts out of the angular.json that had to do with jquery and when I removed this line, with no other changes, everything worked.
> 
> ```
> "node_modules/jqvmap/dist/maps/jquery.vmap.world.js",
> ```

**Comment by chelaruc** [comment]
Hi, @sebas0205! What errors do you get?

**Comment by sushil-suthar** [comment]
I am also facing the same problem in pro version.
Please resolve it ASAP

**Comment by Roman991** [comment]
the collapse component stucks in 'collapsing' state.
I solved this removing jquery-validation and jasny-bootstrap, now it works

**Comment by articaprium** [comment]
Navigation vertical scroll bar is scrolling automatically when user clicks on bottom fields. It should not scroll automatically, in angular. can you help me?

**Comment by rarestoma** [comment]
Hi @articaprium,

Thank you for working with our products.

Can you please help me with a video so I can understand better what happens and find a solution?

Thank you,
Rares

**Comment by racecrew** [comment]
> 
> 
> Hi @chelaruc , I´m still have this problem. I did all the solutions here but nothing .
> I proved #339 and dont work for me
> I proved too and nothing :
> 
> > That did not work for me, still cannot collapse sidebar menus. I have also had jQuery available in the console regardless. You got me thinking though so I started taking different scripts out of the angular.json that had to do with jquery and when I removed this line, with no other changes, everything worked.
> > ```
> > "node_modules/jqvmap/dist/maps/jquery.vmap.world.js",
> > ```

I have still this issue in July 2021 :-( I have tried all the solutuions from this ticket, nothing worked, spent quit much time trying to find out the root-cause.

Finally, the solution was for me taking out from angular.json "node_modules/moment/moment.js", or just moving it to the end of the scripts-array in angular.json.

@all: please, let me know if it worked for you also. Thanks!
